### PR TITLE
Fixed updating Yeti cache

### PR DESCRIPTION
**Pull request by aardschok**
Extended the update logic for the Yeti Cache loader. It was not properly adding or removing the nodes of the container(s).

It is based on the content of the container and the metadata which is published along with the cache. Both the container and the metadata are used to create a lookup with `defaultdict` based on `cbId`. This way we support renaming and adjusting the correct nodes. All IDs which are not in the metadata look up are to be deleted, all new IDs in the metadata file are created and added to the container.

### CODE REVIEW

File: colorbleed/plugins/maya/load/load_yeti_cache.py [hunk 1]
```diff
@@ -83,30 +84,88 @@ def remove(self, container):
...
+
+                for scene_node in scene_nodes:
+                    # Remove the namespace root ":namespace:"
+                    compare_name = scene_node.split(namespace)[-1][1:]
+                    if compare_name != match_name:
+                        self.log.info("Renaming %s to %s" %
+                                      (scene_node, match_name))
+
+                        # Get transform node, this makes renaming easier
+                        transform_node = scene_node.rstrip("Shape")
```
Comment (BigRoy) [inline colorbleed/plugins/maya/load/load_yeti_cache.py]: Let's do `maya.cmds.listRelatives` similar to [here](https://github.com/Colorbleed/colorbleed-config/pull/139/files#diff-e54e80d1c1b4885422ca99bcc6b69d44R124)

File: colorbleed/plugins/maya/load/load_yeti_cache.py [hunk 2]
```diff
@@ -83,30 +84,88 @@ def remove(self, container):
...
+        for cb_id, data in meta_data_lookup.items():
+
+            # Update cache file name
+            file_name = data["name"].replace(":", "_")
+            cache_file_path = "{}.%04d.fur".format(file_name)
+            data["attrs"]["cacheFileName"] = os.path.join(path, cache_file_path)
+
+            if cb_id not in scene_lookup:
+                new_nodes = self.create_nodes(namespace, [data])
+                cmds.sets(*new_nodes, addElement=container_node)
```
Comment (BigRoy) [inline colorbleed/plugins/maya/load/load_yeti_cache.py]: Doesn't this also work without the `*`? Like `cmds.sets(new_nodes, addElement=container_node)`

File: colorbleed/plugins/maya/load/load_yeti_cache.py [hunk 3]
```diff
@@ -152,70 +219,56 @@ def validate_cache(self, filename, pattern="%04d"):
...
+                cache = os.path.join(self.fname,
+                                     "{}.%04d.fur".format(file_name))
+                self.validate_cache(cache)
+                attributes["cacheFileName"] = cache
+
+            # Update attributes with requirements
+            attributes.update({"viewportDensity": 0.1,
+                               "verbosity": 2,
+                               "fileMode": 1,
+                               "visibleInReflections": True,
```
Comment (BigRoy) [inline colorbleed/plugins/maya/load/load_yeti_cache.py]: Nice to see this fixed - but I think it is actually more readable to keep these two "fixes" separate from the attributes update.

How about we do it explicitly like how Sven did it in his PR.

```python
            # Fix for : YETI-6
            # Fixes the render stats (this is literally taken from Perigrene's
            # ../scripts/pgYetiNode.mel script)
            cmds.setAttr(yeti_node + ".visibleInReflections", 1)
            cmds.setAttr(yeti_node + ".visibleInRefractions", 1)
```

File: colorbleed/plugins/maya/load/load_yeti_cache.py (hunk 2 above)
Comment (aardschok) [reply colorbleed/plugins/maya/load/load_yeti_cache.py]: Yes it should, don't know why I used this one :thinking:

**Pull request merged**